keps/sig-storage/177-volume-snapshot/tighten-validation-webhook-crd.md

```diff
@@ -0,0 +1,815 @@
+|  UPDATE   |      spec.Source       |                                                                                                                                                                                                                                    Immutable, no updates allowed.                                                                                                                                                                                                                                     |    400     |
+|  CREATE   | spec.VolumeSnapshotRef | Must have both name and namespace fields set. Preprovisioned: This is the reference to the yet to be created VolumeSnapshot object which should bind to this VolumeSnapshotContent. https://github.com/kubernetes-csi/external-snapshotter/blob/097b1fc7d7cd6576182ca34512c14de1c84b2127/pkg/apis/volumesnapshot/v1beta1/types.go#L270. Dynamic: This is the reference to the VS object which triggered the creation of this VSContent. It also has the UID field, but this is set by the controller. |    400     |
+|  UPDATE   | spec.VolumeSnapshotRef |                                                                                                                                                                                                                                    Immutable, no updates allowed, once it's UID has been set.                                                                                                                                                                                                                                     |    400     |
+
+### Authentication
+
+There are two directions to authentication. Authenticating the webhook server is legitimate, and authenticating the k8s api server is legitimate. 
+
+The API server authenticates the webhook server through TLS certificates and HTTPS. This is required, and an example method of deploying the webhook server with HTTPS will be provided.
+
+Authentication on incoming requests to the webhook server is configurable however out of scope of this document. It’s the user’s responsibility in general to configure the webhook service and the API server if authentication is required(details). The web server implementation, however, should allow users to configure whether authentication is required or not. If no authentication config is specified, the webhook server should default to “NoClientCert”, which effectively will not authenticate the identity of the clients.
```

Comment:
Are there any additional resources about this that we can point to?

Reply:
I added a link to https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#authenticate-apiservers for how to configure the api server to authenticate itself to the webhook server.

There isn't any great documentation on how authenticate the webhook server, however.